Bottom: cb1f2b30deea4f3661533ff076b2fa2b8c14f3f6
Top:    a471dce6bdd2a61c77502759a50c2b1b21812581
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-05-22 14:12:12 -0500

Updates to openldap


---
diff --git a/roles/docker-openldap/tasks/main.yml b/roles/docker-openldap/tasks/main.yml
index e3f23ce..070a494 100644
--- a/roles/docker-openldap/tasks/main.yml
+++ b/roles/docker-openldap/tasks/main.yml
@@ -107,7 +107,7 @@
 # *****************************************************************************
 # Start the data container running
 
-- name: start container (ldap)
+- name: create container (ldap)
   docker_container:
     name: '{{ openldap_container_name }}'
     detach: true
@@ -123,6 +123,19 @@
     image: '{{ openldap_image_name }}:{{ docker_image_tag }}'
     networks:
       - name: '{{ docker_network_frontend }}'
+    state: present
+
+# *****************************************************************************
+# restore?
+
+- include_tasks: restore.yml
+  when: st_openldap_restore.stat.exists == False
+  tags: restore
+
+- name: start container (ldap)
+  docker_container:
+    name: '{{ openldap_container_name }}'
+    state: started
 
 # *****************************************************************************
 # Update or make the image.
@@ -156,22 +169,15 @@
       - name: '{{ docker_network_frontend }}'
     image: '{{ phpldapadmin_image_name }}:{{ docker_image_tag }}'
 
-# *****************************************************************************
-# restore?
-
-- include_tasks: restore.yml
-  when: st_openldap_restore.stat.exists == False
-  tags: restore
-
 # *****************************************************************************
 # Set the LDAP passwords
 
-- name: wait for LDAP to start
-  wait_for:
-    delay: 5
-    host: '{{ container_addr_map.ldap.ip_addr }}'
-    port: 389
-    state: started
+# - name: wait for LDAP to start
+#   wait_for:
+#     delay: 5
+#     host: '{{ openldap_ip_addr }}'
+#     port: 389
+#     state: started
 
 - name: set LDAP admin password
   command: 'docker exec -i {{ openldap_container_name }} /docker-entrypoint.sh set_admin_pass {{ openldap_admin_password | quote }}'
diff --git a/roles/docker-openldap/tasks/restore.yml b/roles/docker-openldap/tasks/restore.yml
index fd82c7f..feaf93e 100644
--- a/roles/docker-openldap/tasks/restore.yml
+++ b/roles/docker-openldap/tasks/restore.yml
@@ -31,14 +31,29 @@
 # restore the openldap backup
 
 - name: stop container (ldap)
-  command: 'docker stop {{ openldap_container_name }}'
+  docker_container:
+    name: '{{ openldap_container_name }}'
+    state: stopped
 
-- name: load ldif files
-  command: 'docker run --rm --volume {{ openldap_cv_name }}:/etc/ldap:z --volume {{ openldap_dv_name }}:/var/lib/ldap:z -v {{ host_openldap_docker_restore_dir }}:/tmp/import_export {{ openldap_image_name }}:{{ docker_image_tag }} apply_ldif -n {{ item.dbnum }} -l {{ item.ldif_file }}'
+- name: restore ldif files
+  docker_container:
+    name: '{{ openldap_container_name }}-restore-{{ item.dbnum }}'
+    detach: false
+    restart_policy: no
+    volumes:
+      - '{{ openssl_dv_name }}:/etc/ssl/private:z'
+      - '{{ openldap_cv_name }}:/etc/ldap:z'
+      - '{{ openldap_dv_name }}:/var/lib/ldap:z'
+      - '{{ openldap_docker_restore_dir }}/{{ item.ldif_file }}:/tmp/import_export/{{ item.ldif_file }}:z'
+    image: '{{ openldap_image_name }}:{{ docker_image_tag }}'
+    command: ['apply_ldif', '-n', '{{ item.dbnum }}', '-l', '{{ item.ldif_file }}']
   with_items: '{{ openldap_backup_files }}'
 
-- name: restart container (ldap)
-  command: 'docker start {{ openldap_container_name }}'
+- name: remove ldif file restoration container
+  docker_container:
+    name: '{{ openldap_container_name }}-restore-{{ item.dbnum }}'
+    state: absent
+  with_items: '{{ openldap_backup_files }}'
 
 # *****************************************************************************
 # cleanup
